test/rubygems/test_gem_package.rb

```diff
@@ -1247,74 +1247,6 @@ def test_verify_truncate
 
   # end #verify tests
 
-  def test_verify_entry
```

Comment:
Having a quick look in `test/rubygems/test_gem_package.rb`, we don't seem to have tests for invalid filenames & filenames with newlines. Could we look to reimplement these cases in our tests? Do you have a recommendation?

Reply:
I replaced this test with a test that manually constructs a tar file that has bad names for the metadata name.  That particular test also had a check that a warning was emitted, but I think we should just change it to a `raise` and allow the normal exception "cause" mechanisms to display the error.

The original introduction of the warning was in ece87d858f018350c53def7a5bc9c810fae73a33, but from what I can tell it's purely for debugging RubyGems itself, and I don't think we should add tests for debug code (we should be adding tests for user facing features / behaviors instead).

Reply:
I put the warning back, I want to address it in a different PR.  If the gem path name is important to the error messages, then we should make it part of the exception message not emit a warning and then _test for the specific warning_.  But I think we should address that in a different patch.